client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoPreparedStatement.java

```diff
@@ -720,6 +728,12 @@ public void setURL(int parameterIndex, URL x)
     public ParameterMetaData getParameterMetaData()
             throws SQLException
     {
+        if (!this.connection().isUseLegacyPreparedStatements()) {
+            try (Statement statement = connection().createStatement()) {
+                statement.execute(format("PREPARE %s FROM %s", statementName, originalSql));
```

Comment:
Let's extract this code into its own method (`prepareStatementIfNecesary()`).  We could also cache the fact that the statement was prepared, so we don't run `PREPARE` multiple times, should metadata methods be invoked more than once.

Reply:
Done